### ISSUE TITLE
Device error in classification tutorial

### ISSUE THREAD

**Issue by KumoLiu** [Project-MONAI/tutorials#912]
**Describe the bug**
`detach` in `decollate_batch` default is True which scalars tensors will be detached into number types instead of torch tensors.

**To reproduce**
```
from monai.transforms import AsDiscrete
device = 'cuda'

y_trans = Compose([AsDiscrete(to_onehot=4)])
raw_label = [torch.tensor([0,1,2]), torch.tensor([1,2,3])]
y = torch.tensor([], dtype=torch.long, device=device)
for i in raw_label:
    y = torch.cat([y, i.to(device)], dim=0)
    
print(y.device) # "cuda"
y_onehot = [y_trans(i) for i in decollate_batch(y)]
# y_onehot = [y_trans(i) for i in decollate_batch(y, detach=False)]
print(y_onehot[0].device) # "cpu"
```
